rpc/handlers.go

```diff
@@ -343,6 +379,70 @@ func (h *Handler) LegacyBlockWithTxs(id BlockID) (*BlockWithTxs, *jsonrpc.Error)
 	return block, nil
 }
 
+func AdaptReceipt(receipt *core.TransactionReceipt, txn core.Transaction,
```

Comment:
What do you think about passing in an rpc.Transaction instead of a core.Transaction? It would save us calling AdaptTransaction inside the function again, and we need to call it outside the function anyway

Reply:
Good point

Reply:
well, there are a few problems that I faced trying to do that now:
* `feeUnit` already used in a place where only core.Transaction is used (there is no need to call AdaptTransaction on caller side)
* `v.MessageHash()` for L1Handler type not implemented on rpc.Transaction type 

I'll add a comment to try restruct passed data